Add py-atlinter version 0.2.3

Stannislav opened the pull request:
* Starting with version `atlinter==0.2.3` the package is distributed via GitHub/PyPI (open-sourced)
* Pulled `mxnet`, `opencv`, and `py-graphviz` from upstream to make builds work
* Adjusted version ranges in pulled packages for the non-inclusive range syntax
* Re-formatted the pulled `opencv` package to make `flake8` tests pass.
* Pulled `onnx` from upstream after unit tests failed:
  ```text
  E       AssertionError: These packages have missing dependencies:
  E         opencv: [onnx]
  E       assert not {'opencv': {'onnx'}}
  ```

Commit:
Switch to double quotes

Commit:
Add py-atlinter versions up to 0.2.3

Commit:
Pull the "mxnet" package from upstream

Commit:
Pull the opencv package from upstream

Commit:
Fix version range syntax

Commit:
Correctly specify mxnet's variants

Commit:
Pull py-graphviz from upstream

Commit:
Restore the url field in py-graphviz

Commit:
Add "build" to type of python dependencies

Commit:
Fix code style (flake8)

Commit:
Pull the onnx package from upstream

Commit:
Fix code style in onnx

Stannislav commented:
Help needed to debug the build failure.

Locally `spack install py-atlinter` works.

Maybe the packages pulled from upstream are causing problems elsewhere?

matz-e commented on var/spack/repos/builtin/packages/opencv/package.py:
```diff
@@ -1,364 +1,406 @@
+        'apps', 'calib3d', 'core', 'dnn', 'features2d', 'flann', 'gapi',
+        'highgui', 'imgcodecs', 'imgproc', 'java', 'java_bindings_generator',
+        'js', 'js_bindings_generator', 'ml', 'objc', 'objc_bindings_generator',
+        'objdetect', 'photo', 'python2', 'python3',
```
I meant before to update from upstream what is essential. Were all the bumps needed?

Anyways, there is now no `+python` variant, but a `+python3` variant. You'll have to change the ML packages from `opencv+python` to `opencv+python3` (IMO splitting the variant like that was a bad decision…)

Stannislav replied:
About the bumps, I started with pulling `mxnet`, and after that more things stopped working so I pulled more. The dependencies can be summaried like that:
```
py-atlinter
└── mxnet
    ├── opencv
    │   └── onnx
    └── py-graphvz
```
Maybe there was a way of minimally adjusting `mxnet` by hand without pulling everything from upstream.

Stannislav replied:
Re `+python`, nice find. I grepped to find where this could happend and I'm seeing this:
```
var/spack/repos/builtin/packages/candle-benchmarks/package.py:28:    depends_on('opencv@3.2.0: +core +highgui +imgproc +jpeg +png +tiff +zlib +python -dnn ~eigen ~gtk')
var/spack/repos/builtin/packages/py-cinema-lib/package.py:40:    depends_on('opencv@3.4:+python', when='+opencv', type=('build', 'run'))
var/spack/repos/builtin/packages/py-imgaug/package.py:28:    depends_on('opencv +python', type=('run'))
var/spack/repos/builtin/packages/py-mmcv/package.py:23:    depends_on('opencv+python', type=('build', 'run'))
...
```
I guess these would all have problems with `opencv` lacking the `+python` variant?

matz-e replied:
Yes. I'd only change the ones you use, to avoid merge conflicts down the line.

And I'd rather see the bumps in the dependencies than having to manually merge conflicts in `mxnet` later :-/

matz-e reviewed (commented):
So…

GianlucaFicarelli committed:
Fix py-bluepysnap dependencies (#1355)

sergiorg-hpc committed:
Fix for incorrect tag selection in Steps recipe (#1356)

pramodk committed:
Use HPE-MPI v2.25 (1.9 release) for deployment (#1320)

We have observed performance regression and deadlocks at scale with versions <= v2.23. This new version v2.25 (1.9 release) solves these issues and includes a bug fix that re-enables support for IME in MPI-IO.

alkino committed:
Catch2: update with last version (#1353)

lguyot committed:
deployment: py-atlas-building-tools 0.1.6 and its dependency py-cgal-pyding 0.1.2 [BBPP82-630] (#1357)

lguyot committed:
(deployment): Fixe latest version of py-atlas-building-tools (#1358)

matz-e committed:
touchdetector: does not actually need HighFive (#1359)

matz-e committed:
morpho-kit: update (#1362)

matz-e committed:
touchdetector: drop boost dependency (#1363)

st4rl3ss committed:
Added new versions of SpatialIndex (#1360)

* Added new tags for SpatialIndex

Commit:
Undo pulls from upstream

Commit:
Merge branch 'develop' into upgrade-py-atlinter-0.2.3

Commit:
Disable the opencv varian in mxnet dep

Stannislav commented:
Moving the discussion here from slack.

The tests all pass, but when testing locally, after `spack load py-atlinter@0.2.3` I get
```
-bash: /usr/bin/env: Argument list too long
```
after runnung  _any_ command in the shell. According to [a thread on SO](https://stackoverflow.com/a/26270155/2804645) it's likely that the environment uses too much space.

What's the reason? Here's the module file:

<details><summary>./spack-software/modules/tcl/linux-rhel7-x86_64/py-atlinter/0.2.3-6ssgmc</summary>

```
#%Module1.0
## Module file created by spack (https://github.com/spack/spack) on 2021-11-16 10:55:27.776140
##
## py-atlinter@0.2.3%gcc@9.3.0~cuda arch=linux-rhel7-x86_64/6ssgmc4
##


module-whatis "Interpolation of section images."

proc ModulesHelp { } {
puts stderr "Interpolation of section images."
}

if { [ module-info mode load ] && ![ is-loaded python/3.8.3-qxs7he ] } {
    puts stderr "Autoloading python/3.8.3-qxs7he"
    module load python/3.8.3-qxs7he
}
if { [ module-info mode load ] && ![ is-loaded py-statsmodels/0.10.2-go7tqc ] } {
    puts stderr "Autoloading py-statsmodels/0.10.2-go7tqc"
    module load py-statsmodels/0.10.2-go7tqc
}
if { [ module-info mode load ] && ![ is-loaded py-scikit-learn/0.23.2-65ozrc ] } {
    puts stderr "Autoloading py-scikit-learn/0.23.2-65ozrc"
    module load py-scikit-learn/0.23.2-65ozrc
}
if { [ module-info mode load ] && ![ is-loaded py-patsy/0.5.1-cj4ipa ] } {
    puts stderr "Autoloading py-patsy/0.5.1-cj4ipa"
    module load py-patsy/0.5.1-cj4ipa
}
if { [ module-info mode load ] && ![ is-loaded py-opencv-python/4.5.3.56-tpwchg ] } {
    puts stderr "Autoloading py-opencv-python/4.5.3.56-tpwchg"
    module load py-opencv-python/4.5.3.56-tpwchg
}
if { [ module-info mode load ] && ![ is-loaded py-numpy/1.19.4-upzqna ] } {
    puts stderr "Autoloading py-numpy/1.19.4-upzqna"
    module load py-numpy/1.19.4-upzqna
}
if { [ module-info mode load ] && ![ is-loaded py-notebook/6.1.4-e2dh4o ] } {
    puts stderr "Autoloading py-notebook/6.1.4-e2dh4o"
    module load py-notebook/6.1.4-e2dh4o
}
if { [ module-info mode load ] && ![ is-loaded py-nibabel/3.2.1-n72zha ] } {
    puts stderr "Autoloading py-nibabel/3.2.1-n72zha"
    module load py-nibabel/3.2.1-n72zha
}
if { [ module-info mode load ] && ![ is-loaded py-chart-studio/1.1.0-vjaneu ] } {
    puts stderr "Autoloading py-chart-studio/1.1.0-vjaneu"
    module load py-chart-studio/1.1.0-vjaneu
}
if { [ module-info mode load ] && ![ is-loaded hpe-mpi/2.25.hmpt-ll3i6f ] } {
    puts stderr "Autoloading hpe-mpi/2.25.hmpt-ll3i6f"
    module load hpe-mpi/2.25.hmpt-ll3i6f
}
if { [ module-info mode load ] && ![ is-loaded hdf5/1.10.7-z5dvzg ] } {
    puts stderr "Autoloading hdf5/1.10.7-z5dvzg"
    module load hdf5/1.10.7-z5dvzg
}
if { [ module-info mode load ] && ![ is-loaded ffmpeg/4.2.2-qhfrsy ] } {
    puts stderr "Autoloading ffmpeg/4.2.2-qhfrsy"
    module load ffmpeg/4.2.2-qhfrsy
}

prepend-path LD_LIBRARY_PATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-atlinter-0.2.3-6ssgmc/lib"
prepend-path LIBRARY_PATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-atlinter-0.2.3-6ssgmc/lib"
prepend-path CMAKE_PREFIX_PATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-atlinter-0.2.3-6ssgmc/"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-atlinter-0.2.3-6ssgmc/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-scikit-learn-0.23.2-65ozrc/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-bottleneck-1.2.1-6sicuo/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pygments-2.6.1-qlxrtl/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-python-dateutil-2.8.0-dyug57/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-atlannot-0.1.2-dh565o/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/applications/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-webcolors-1.11.1-kn5u7n/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-joblib-1.0.1-k5fdw7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pybind11-2.5.0-k7lp43/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-cycler-0.10.0-awhu2s/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-atlinter-0.2.3-6ssgmc/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-jsonschema-3.2.0-iy2mwu/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pandas-1.2.5-cnanak/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-widgetsnbextension-3.5.1-5jvi5x/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-setuptools-50.3.2-afe2g4/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pickleshare-0.7.5-gf4xh7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pytz-2020.1-d6oa2i/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pyzmq-19.0.0-tllcka/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-threadpoolctl-2.0.0-vzgkxs/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-chardet-3.0.4-42rahz/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-ipywidgets-7.5.1-tme7rh/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pandocfilters-1.4.2-ebpwug/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-packaging-19.2-qavb3u/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-certifi-2020.6.20-viqjvd/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-torch-1.7.0-fm4w7a/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-opencv-python-4.5.3.56-tpwchg/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/mxnet-1.8.0-sjofkz/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-notebook-6.1.4-e2dh4o/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-idna-2.8-kxxezr/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-nbformat-5.0.7-s3czgk/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-future-0.18.2-kpddcv/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/applications/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-scikit-image-0.17.2-xprycg/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-traitlets-5.0.4-okigg6/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-statsmodels-0.10.2-go7tqc/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-requests-2.24.0-h3w463/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-nest-asyncio-1.5.1-kdaoa7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-argon2-cffi-20.1.0-7w46is/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-prompt-toolkit-2.0.9-ufy4mr/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pyyaml-5.3.1-5wd4ic/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-atldld-0.2.2-7f2qzi/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-prometheus-client-0.7.1-zzyfxh/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pyrsistent-0.16.0-psjavy/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/applications/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-tifffile-2020.10.1-bl5wyz/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pycparser-2.20-qmoqiq/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pillow-8.0.0-gprprf/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-pytorch-fid-0.2.0-m74nio/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-ipython-genutils-0.2.0-xso2c7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-nbconvert-6.0.1-yv4bgg/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-cffi-1.14.3-xr5fxv/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-entrypoints-0.3-5fuziq/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-jupyterlab-pygments-0.1.1-pi2xh7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-kiwisolver-1.1.0-mjc7tm/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-jupyter-client-6.1.7-aq6oby/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-py-1.8.0-p4ixtm/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/applications/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pywavelets-1.1.1-yommiy/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-patsy-0.5.1-cj4ipa/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-parso-0.6.1-k37onn/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-send2trash-1.5.0-g6wlkt/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-antspyx-0.2.4-z7egeu/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pexpect-4.7.0-tx666n/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-nbclient-0.5.0-qpn3t7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-terminado-0.8.3-gb3gpm/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-urllib3-1.25.6-hvsjls/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-chart-studio-1.1.0-vjaneu/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-testpath-0.4.2-of65kv/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/applications/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pynrrd-0.4.0-kapfjq/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-pyparsing-2.4.2-lzgoob/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/applications/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-matplotlib-3.3.3-wajzco/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-jedi-0.13.3-fzoyi7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-ipykernel-5.3.4-csebe4/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-retrying-1.3.3-gibbwi/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-nibabel-3.2.1-n72zha/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-cython-0.29.21-6s4srk/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/applications/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-imageio-2.5.0-5r26ka/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-typing-extensions-3.10.0.0-lv47gz/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-backcall-0.2.0-dchwa5/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-async-generator-1.10-xg5f4k/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-scipy-1.5.4-t6m7lq/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-numpy-1.19.4-upzqna/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-six-1.16.0-wp4e4n/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-mistune-0.8.4-xws3ve/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-decorator-4.4.2-njywgr/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-defusedxml-0.6.0-bcz5c7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-ipython-7.18.1-tkuz3o/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-torchvision-0.8.1-fn5fd4/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-babel-2.7.0-lb34zm/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-wcwidth-0.1.7-4iuudp/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-toml-0.10.0-hdgffv/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-ptyprocess-0.6.0-jbrrox/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/applications/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-networkx-2.4-evadrc/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-tornado-6.0.3-io6dz7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-jupyter-core-4.6.3-pt5bxf/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-bleach-3.1.0-xs5hw7/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-webencodings-0.5.1-vt4hzl/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-plotly-4.1.0-uy7pgm/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-attrs-20.3.0-ldgbmw/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-numexpr-2.7.0-awm27s/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-tqdm-4.59.0-jq4aua/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-markupsafe-1.1.1-ro7q4k/lib/python3.8/site-packages"
prepend-path PYTHONPATH "/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-jinja2-2.10.3-c3pik7/lib/python3.8/site-packages"
setenv PY_ATLINTER_ROOT "/gpfs/bbp.cscs.ch/home/sschmidt/.local/spack-software/linux-rhel7-x86_64/gcc-9.3.0/py-atlinter-0.2.3-6ssgmc"
```
</details>

According to @matz-e 

> I think all those "autoload py-*" are screwing with things, too

So is there anythink I can do about this? I'm afraid I'm too unfamiliar with the spack internals to answer...

matz-e commented:
> Moving the discussion here from slack.
> 
> The tests all pass, but when testing locally, after `spack load py-atlinter@0.2.3` I get
> 
> ```
> -bash: /usr/bin/env: Argument list too long
> ```

Stupid question: do you run any fancy prompts?

I'm "testing" the PR like this:
```
matwolf in bbpv2 in ~ via  v14.13.0 via 🐍 v2.7.5 
❯ unset MODULEPATH

matwolf in bbpv2 in ~ via  v14.13.0 via 🐍 v2.7.5 
❯ . /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1354/config/modules.sh 

matwolf in bbpv2 in ~ via  v14.13.0 via 🐍 v2.7.5 
❯ module load unstable           

matwolf in bbpv2 in ~ via  v14.13.0 via 🐍 v2.7.5 
❯ module av py-atl            

------------------------ /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1354/modules/applications/2021-11/tcl/linux-rhel7-x86_64 ------------------------
py-atlalign/0.6.0.20210902    py-atlannot/0.1.2             py-atlas-building-tools/0.1.7 py-atldld/0.3.2               py-atlinter/0.2.3

matwolf in bbpv2 in ~ via  v14.13.0 via 🐍 v2.7.5 
❯ module load py-atlinter/0.2.3
Autoloading python/3.8.3
Autoloading py-atlannot/0.1.2
Autoloading ffmpeg/4.2.2

matwolf in bbpv2 in ~ via  v14.13.0 via 🐍 v3.8.3 
❯ ls                           
スタフ                   other_files
```
See: https://github.com/BlueBrain/spack/blob/develop/deploy/docs/deploy_bb5.md#testing-pull-requests-for-updates

> > I think all those "autoload py-*" are screwing with things, too
> 
> So is there anythink I can do about this? I'm afraid I'm too unfamiliar with the spack internals to answer...

Do you whitelist everything for modules? Don't do that… only whitelist what you really want.

Stannislav commented:
@matz-e 
> Stupid question: do you run any fancy prompts?

Not really, just the default `/etc/bash` with a custom `PS1`, I hope that's not fancy.

> I'm "testing" the PR like this: [...]

Thanks I'm going to try this. I usually test on the build that I run myself and not the one from Jenkins, something like this:
```bash
spack install py-atlinter@0.2.3
spack load py-atlinter@0.2.3
```
But a good point that if it works with the CI build then it should be fine. I'll give it a try and update here later.

---
EDIT: @matz-e just tested with your commands and it all works.

Stannislav commented:
> Do you whitelist everything for modules? Don't do that… only whitelist what you really want.

Following [this readme](https://github.com/BlueBrain/spack/blob/develop/deploy/docs/setup_bb5.md#automatically-generate-modules-for-all-installed-software) I have `${HOME}/.spack/modules.yaml` file that looks like this:
```yaml
modules:
  tcl:
    whitelist:
      - '@:'
    projections:
      all: '{name}/{version}-{hash:6}'
```
Is that what you mean by whitelisting all modules? Should I not do that?

matz-e commented:
> ```yaml
> modules:
>   tcl:
>     whitelist:
>       - '@:'
>     projections:
>       all: '{name}/{version}-{hash:6}'
> ```
> 
> Is that what you mean by whitelisting all modules? Should I not do that?

I don't think it's a good practice, some folks insist on doing it, though :shrug: 

I'd rather explicitly add the top-level software I build in there. Helps keep the list of modules down to a manageable level, too.

matz-e commented:
> EDIT: @matz-e just tested with your commands and it all works.

Feel free to merge then! :tada: :taco: :beers:

Stannislav commented:
> > EDIT: @matz-e just tested with your commands and it all works.
> 
> Feel free to merge then! 🎉 🌮 🍻

Awesome, thanks a lot, appreciate your help.
